x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -325,8 +319,18 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
             return noScaleResultOrRefresh(reasonBuilder, memoryTrackingStale, new AutoscalingDeciderResult(
                 context.currentCapacity(),
                 reasonBuilder
-                    .setSimpleReason("Passing currently perceived capacity as there are analytics and anomaly jobs in the queue, " +
-                        "but the number in the queue is less than the configured maximum allowed.")
+                    .setSimpleReason(
+                        String.format(
```

Comment:
Should we rather use `ParameterizedMessage` here?
How do we decide whether to use `ParameterizedMessage` or `String.format` in general?

Reply:
I use `ParameterizedMessage` when I need a message supplier or a "Message" object. This is the case in logging.

I use String.format (or a similar message format that we use in the ML plugin). 

Both are used in the code base, I don't think one is preferred over the other.